sirepo/http_util.py

```diff
@@ -0,0 +1,48 @@
+"""Support routines for http requests/responses
+
+:copyright: Copyright (c) 2024 RadiaSoft LLC.  All Rights Reserved.
+:license: http://www.apache.org/licenses/LICENSE-2.0.html
+"""
+
+from pykern.pkcollections import PKDict
+from pykern.pkdebug import pkdlog
+import re
+import sirepo.job
+
+#: Http auth header name
+_AUTH_HEADER = "Authorization"
+
+#: http auth header scheme bearer
+_AUTH_HEADER_SCHEME_BEARER = "Bearer"
+
+#: Regex to test format of auth header and extract token
+_AUTH_HEADER_RE = re.compile(
+    _AUTH_HEADER_SCHEME_BEARER + r"\s(" + sirepo.job.UNIQUE_KEY_CHARS_RE + ")",
+    re.IGNORECASE,
+)
+
+
+def auth_header(token):
+    """Construct auth header object.
+
+    Args:
+      token (str): Secret to be included in the header
+    Returns
+      PKDict: auth header
+    """
+    return PKDict({_AUTH_HEADER: f"{_AUTH_HEADER_SCHEME_BEARER} {token}"})
+
+
+def parse_auth_header(headers):
+    """Validate and retrieve authentication header.
```

Comment:
`Parase and retrieve RFC6750 bearer token

Reply:
390567377